### PR TITLE
Initial merge from original drewandersonnz repo

### PR DESCRIPTION
**Pull request by drewandersonnz**
Multiple people have committed to this repo, this PR is to bring the commit history of multiple contributors forward.

### CODE REVIEW

**Comment by drewandersonnz** [comment]
Should this be one squashed commit? Or maintain the commit history?

@georgettica suggested this: https://stackoverflow.com/questions/7442112/how-to-attribute-a-single-commit-to-multiple-developers

**Comment by 2uasimojo** [comment]
Personally I prefer leaving the individual commits.

**Comment by karthikperu7** [comment]
I would vote for keeping the individual commits too.

File: container-setup/install/install-kube_ps1.sh
Comment (clcollins) [inline container-setup/install/install-kube_ps1.sh:12]: This should pull either directly from the upstream kube-ps1, or use a copy forked in this repo, just in case (heaven forbid) Drew decides to leave us someday.

File: Dockerfile
Comment (clcollins) [inline Dockerfile:2]: This should probably be a tagged image version that we consciously update when a new version comes out.  I've previously had bad experiences when "latest" changed to point to a newer version and some things broke.  We should ensure that the Dockerfile is in a working tested state, and we can't do that when "latest" can arbitrarily point to a new version.

File: Dockerfile
Comment (clcollins) [inline Dockerfile:6]: `yum` is deprecated/deprecating in current versions of Fedora .  Perhaps change to `dnf`?

File: Dockerfile
Comment (clcollins) [inline Dockerfile:19]: You can save more disk space by removing the yum cache as well:

`&& yum clean all && rm -r /var/cache/yum/`

File: container-setup/install/install-aws.sh
Comment (clcollins) [inline container-setup/install/install-aws.sh:3]: What is the benefit of this? To prevent accidental install?  Could this just be a non-executable file that is set to executable when copied into the container instead?  This just adds extra complexity to all the files and the Dockerfile/is not DRY.

File: container-setup/install/install-oc.sh
Comment (clcollins) [inline container-setup/install/install-oc.sh:3]: Ditto above.

File: container-setup/install/install-osdctl.sh
Comment (clcollins) [inline container-setup/install/install-osdctl.sh:3]: Ditto above.

File: container-setup/install/install-rosa.sh
Comment (clcollins) [inline container-setup/install/install-rosa.sh:3]: Ditto

File: container-setup/install/install-rosa.sh
Comment (clcollins) [inline container-setup/install/install-rosa.sh:8]: I might be missing something, but is this doing anything?  The only function in there is `remove_coloring()` and I don't see it called/used in here.

File: container-setup/install/install-utils.sh
Comment (clcollins) [inline container-setup/install/install-utils.sh:3]: Ditto

File: container-setup/install/install-utils.sh
Comment (clcollins) [inline container-setup/install/install-utils.sh:12]: If this is running inside a container, designed to be run as root anyway, why not just put the utilities in one of the normal $PATH bin directories?  Or use ~/.local/bin rather than a custom dir?  This file could be replaced with one line in the Dockerfile, maybe.

File: container-setup/install/install-velero.sh
Comment (clcollins) [inline container-setup/install/install-velero.sh:3]: Ditto

**Review by clcollins — APPROVED** [review]
This looks fine to me for an initial commit into the repo. There is nothing internally sensitive included and it's working great.

I added several comments/review notes, but those were mostly as a reminder for things I want to address myself later if someone doesn't have a quick fix.

/ltgm
/approve

**Comment by rogbas** [comment]
Overall it LGTM, we could address the comments from this review in a separate PR

**Comment by clcollins** [comment]
> Overall it LGTM, we could address the comments from this review in a separate PR

@rogbas That was my intention.  Just wanted to get them noted down.

File: Dockerfile
Comment (iamkirkbater) [reply Dockerfile:2]: I have mixed feelings about it.  On one hand, we have issues if they update the upstream.  On the other hand, we have to have someone watch the upstream and update our build when necessary.

For a project like this, I think it's safe to stay on latest but be cognizant of that, especially once we get to the point where you won't be building this locally unless you're heavily customizing it or if you're building the release, in which case we'd probably want the latest updated at that point anyway.

With the end goal of providing a quay.io image that the average user of ocm-container can just download and run, as long as we tag _those_ appropriately I'm in the camp of leaving this at :latest, but if I'm the minority I'll concede the argument.

File: Dockerfile
Comment (iamkirkbater) [reply Dockerfile:6]: I believe we are going to leave `fedora` as a base image anyway and either move to coreos container linux or centos as our base, but I don't think we've finalized on which we are moving towards.  Good to keep in mind though if something breaks before we make the migration.

File: Dockerfile
Comment (iamkirkbater) [reply Dockerfile:19]: Nice, we should add that in!

```suggestion
    && yum clean all \
    && rm -r /var/cache/yum;
```

File: container-setup/install/install-aws.sh
Comment (iamkirkbater) [reply container-setup/install/install-aws.sh:3]: Ooohhh, I really like that idea.  I'll card that up.

File: container-setup/install/install-rosa.sh
Comment (iamkirkbater) [reply container-setup/install/install-rosa.sh:8]: You're probably right.  ROSA used to be installed via `go get` I believe and that used to dirty up the terminal (because for whatever reason the red color would persist through the rest of the build).  It should be safe to remove this when we move to including only the binaries instead of the entire source repos.

File: container-setup/install/install-utils.sh
Comment (iamkirkbater) [reply container-setup/install/install-utils.sh:12]: Funny you mention this, I actually had to make some edits to this to get `docker exec` to work and auto-login as well as to be able to run commands from outside the container.

I ended up changing the path in the Dockerfile and removing this in another PR that will come up here after this one merges.

File: Dockerfile
Comment (clcollins) [reply Dockerfile:2]: I would argue you have to watch something anyway.  You already have versions of the binaries hard-coded, so at the point they need to be upgraded, a human has to increment a number, just like this.

If you were to keep `latest` and switch to using tests to verify, then you need a human watching the build tests, unless PRs fail to build.  I would say as long as each PR is built as part of it's testing, then that is probably the best solution:

* leave as latest
* move other binaries to latest
* make each PR test require an image build

File: Dockerfile
Comment (clcollins) [reply Dockerfile:6]: :+1:

File: container-setup/install/install-kube_ps1.sh
Comment (drewandersonnz) [reply container-setup/install/install-kube_ps1.sh:12]: I agree with having it not tied to me, it is a fork in my repo so we can check any changes ... I don't want to reply on someone external to our team for a security review of inbound code.

Last thing we want is an attacker gaining access to and submitting dangerous code to a smaller repo that we depend on.

I'm less concerned about other tools in the `github.com/openshift` organisation, as we are in control of that.

File: Dockerfile
Comment (drewandersonnz) [reply Dockerfile:2]: There is a consideration moving to automation for image builds and tests. This should alert us to problems as they happen.

File: Dockerfile
Comment (drewandersonnz) [reply Dockerfile:6]: I think it might be good to deploy multiple base image versions... `ocm-container:fedora`, `om-container:ubuntu`, `ocm-container:alpine` for example. 

As we use it as an interactive environment, people shouldn't be made to learn tools they're not familiar with.

File: container-setup/install/install-aws.sh
Comment (drewandersonnz) [reply container-setup/install/install-aws.sh:3]: Yes, intention is to block execution on your local machine, in case of accidental execution outside of container. +1, I like the "not executable until required".

File: container-setup/install/install-kube_ps1.sh
Comment (drewandersonnz) [reply container-setup/install/install-kube_ps1.sh:12]: As above, this fork is a safety-net, we are not using upstream directly to be able to review any code additions. This could be vendored, or cloned to openshift org, or some other thing. But some link through to finally upstream would be good for recognition of that projects contributions, and for being able to stay in sync.

**Pull request merged**
